### PR TITLE
dialects: Implement affine.parallel

### PR DESCRIPTION
**Pull request by PapyChacal**

### CODE REVIEW

File: xdsl/dialects/affine.py [hunk 1]
```diff
@@ -142,6 +146,41 @@ class If(IRDLOperation):
     else_region = region_def()
 
 
+@irdl_op_definition
+class ParallelOp(IRDLOperation):
```
Comment (superlopuh) [inline xdsl/dialects/affine.py:150]: nit: link to doc in MLIR?

**Review by superlopuh — COMMENTED** [review]
Could you add an interop test, please?

File: xdsl/dialects/affine.py (hunk 1 above)
Comment (PapyChacal) [reply xdsl/dialects/affine.py:150]: Is that a new convention we have? Otherwise, why specifically here?

**Comment by PapyChacal** [comment]
> Could you add an interop test, please?

#1981 ! Will rebase this one, nothing urgent

File: xdsl/dialects/affine.py (hunk 1 above)
Comment (superlopuh) [reply xdsl/dialects/affine.py:150]: No, I just felt like it would be useful here specifically, don't see a downside to it also

**Review by superlopuh — APPROVED** [review]
🚀

**Pull request merged**